### PR TITLE
adds timeout status condition primitives

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3708. Also see https://github.com/giantswarm/node-operator/pull/51.

### CODE REVIEW

File: pkg/apis/core/v1alpha1/drainer_funcs.go [hunk 1]
```diff
@@ -1,23 +1,33 @@
 package v1alpha1
 
-const (
-	DrainerConfigStatusStatusTrue  = "True"
-	DrainerConfigStatusTypeDrained = "Drained"
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/drainer_funcs.go]: Moved to types.

File: pkg/apis/core/v1alpha1/drainer_types.go [hunk 2]
```diff
@@ -7,6 +7,18 @@ import (
...
+	DrainerConfigStatusStatusTrue = "True"
+)
+
+const (
+	DrainerConfigStatusTypeDrained = "Drained"
+)
+
+const (
+	DrainerConfigStatusTypeTimeout = "Timeout"
+)
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/drainer_types.go]: This is the new condition.

File: pkg/apis/provider/v1alpha1/status_funcs.go [hunk 3]
```diff
@@ -52,9 +52,9 @@ func (s StatusCluster) WithUpdatingCondition() []StatusClusterCondition {
 	return withCondition(s.Conditions, StatusClusterTypeUpdated, StatusClusterTypeUpdating, StatusClusterStatusTrue)
 }
 
-func hasCondition(conditions []StatusClusterCondition, search string, status string) bool {
+func hasCondition(conditions []StatusClusterCondition, s string, t string) bool {
 	for _, c := range conditions {
-		if c.Status == search && c.Type == status {
+		if c.Status == s && c.Type == t {
```
Comment (xh3b4sd) [inline pkg/apis/provider/v1alpha1/status_funcs.go]: Fixing the naming here.

**Review by MarcelMue — APPROVED** [review]
LGTM :+1:

**Pull request merged**